src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1030,6 +1030,51 @@ public void Connect(IPAddress[] addresses, int port)
             }
         }
 
+        public void Close()
+        {
+            if (s_loggingEnabled)
+            {
+                NetEventSource.Enter(NetEventSource.ComponentType.Socket, this, nameof(Close), null);
+            }
+            if (GlobalLog.IsEnabled)
+            {
+                GlobalLog.Print("Socket#" + LoggingHash.HashString(this) + "::Close() timeout = " + _closeTimeout);
+            }
+
+            Dispose();
+
+            if (s_loggingEnabled)
+            {
+                NetEventSource.Exit(NetEventSource.ComponentType.Socket, this, nameof(Close), null);
+            }
+        }
```

Comment:
Nit: I wonder whether we need all of this additional logging here, whether this could just be: ```public void Close() => Dispose();```.  But if this is what desktop has and the goal here is keeping it the same, that's fine, too.